vignettes/S4VectorOverview.Rmd

```diff
@@ -0,0 +1,369 @@
+
+1.`window` - Extracts a subsequence over a specified region.
+
+2.`subset` - Extracts the subsequence specified by a logical vector.
+
+3.`head` - Extracts a consecutive subsequence containing the first n
+elements.
+
+4.`tail` - Extracts a consecutive subsequence containing the last n
+elements.
+
+5.`rev` - Creates a new sequence with the elements in the reverse order.
+
+6.`rep` - Creates a new sequence by repeating sequence elements.
+
+The following code illustrates how these functions are used on an *Rle*
+vector:
+
+```{r seq-extraction}
+xSnippet <- window(xRle, 4751, 4760)
+xSnippet
+```
+
+```{r}
+head(xSnippet)
+```
+
+```{r}
+tail(xSnippet)
+```
+
+```{r}
+rev(xSnippet)
+```
+
+```{r}
+rep(xSnippet, 2)
+```
+
+```{r}
+subset(xSnippet, xSnippet >= 5L)
+```
+
+### Concatenating vector-like objects
+
+The
+[*S4Vectors*](http://bioconductor.org/packages/release/bioc/html/S4Vectors.html)
+package uses two generic functions, `c` and `append`, for concatenating
+two *Vector* derivatives. The methods for *Vector* objects follow the
+definition that these two functions are given the
+[*base*](http://bioconductor.org/about/removed-packages/) package.
+
+```{r seq-concatenate}
+c(xSnippet, rev(xSnippet))
+```
+
+```{r}
+append(xSnippet, xSnippet, after=3)
+```
+
+### Looping over subsequences of vector-like objects
+
+In *R*, \Rfunction{for} looping can be an expensive operation. To
+compensate for this,
+the[*S4Vectors*](http://bioconductor.org/packages/release/bioc/html/S4Vectors.html)
+package provides `aggregate` and `shiftApply` methods `shiftApply` is a
+new generic function defined in
+[*S4Vectors*](http://bioconductor.org/packages/release/bioc/html/S4Vectors.html))
+to perform calculations over subsequences of vector-like objects.
+
+The `aggregate` function combines sequence extraction functionality of
+the `window` function with looping capabilities of the `sapply`
+function. For example, here is some code to compute medians across a
+moving window of width 3 using the function `aggregate`:
+
+```{r aggregate}
+xSnippet
+```
+
+```{r}
+aggregate(xSnippet, start=1:8, width=3, FUN=median)
+```
+
+The `shiftApply` function is a looping operation involving two
+vector-like objects whose elements are lined up via a positional shift
+operation. For example, the elements of `xRle` and `yRle` were simulated
+from Poisson distributions with the mean of element i from `yRle` being
+equivalent to the mean of element i + 250 from `xRle`. If we did not
+know the size of the shift, we could estimate it by finding the shift
+that maximizes the correlation between `xRle` and `yRle`.
+
+```{r shiftApply-cor}
+cor(xRle, yRle)
+```
+
+```{r}
+shifts <- seq(235, 265, by=3)
+corrs  <- shiftApply(shifts, yRle, xRle, FUN=cor)
+```
+
+```{r figshiftcorrs, fig=TRUE, eps=FALSE, width=5,fig.align = 'center', height=5, fig.cap='Correlation between *xRle* and *yRle* for various shifts'}
+plot(shifts, corrs)

```

Comment:
Did not find a way to move it since we only get the plot after the plot command. But if I find an easy way to do it I will but for now it will just remain there as you have suggested.